src/main/java/com/palantir/gradle/versions/lockstate/LockStates.java

```diff
@@ -99,6 +102,16 @@ private static Line componentWithDependentsToLine(ModuleVersionIdentifier compon
                 .collect(toList());
     }
 
+    private static String versionConstraintToString(VersionConstraint versionConstraint) {
+        String constraintString = versionConstraint.toString();
+
+        if (SINGLE_VERSION_RANGE.matcher(constraintString).matches()) {
```

Comment:
does this match the whole string necessarily?

Reply:
`.matches()` does yes. (`.find()` doesn't):

> Attempts to match the entire region against the pattern.